tests/topotests/bgp_vrf_ipv6_link_local/test_bgp_vrf_ipv6_ll_func.py

```diff
@@ -0,0 +1,206 @@
+    write_test_footer,
+    check_address_types,
+    apply_raw_config,
+    required_linux_kernel_version,
+)
+from lib.topolog import logger
+from lib.bgp import (
+    verify_bgp_convergence,
+    verify_bgp_convergence_from_running_config,
+    create_router_bgp,
+)
+from lib.topojson import build_topo_from_json, build_config_from_json
+
+# Reading the data from JSON File for topology creation
+jsonFile = "{}/bgp_vrf_ipv6_ll_func.json".format(CWD)
+try:
+    with open(jsonFile, "r") as topoJson:
+        topo = json.load(topoJson)
+except IOError:
+    assert False, "Could not read file {}".format(jsonFile)
+
+# Global Variable
+KEEPALIVETIMER = 2
+HOLDDOWNTIMER = 6
+
+
+class CreateTopo(Topo):
+    """
+    Test BasicTopo - topology 1
+
+    * `Topo`: Topology object
+    """
+
+    def build(self, *_args, **_opts):
+        """Build function"""
+        tgen = get_topogen(self)
+
+        # Building topology from json file
+        build_topo_from_json(tgen, topo)
+
+
+def setup_module(mod):
+    """
+    Sets up the pytest environment
+
+    * `mod`: module name
+    """
+
+    # Required linux kernel version for this suite to run.
+    result = required_linux_kernel_version("4.15")
+    if result is not True:
+        pytest.skip("Kernel requirements are not met")
+
+    testsuite_run_time = time.asctime(time.localtime(time.time()))
+    logger.info("Testsuite start time: {}".format(testsuite_run_time))
+    logger.info("=" * 40)
+
+    logger.info("Running setup_module to create topology")
+
+    # This function initiates the topology build with Topogen...
+    tgen = Topogen(CreateTopo, mod.__name__)
+    # ... and here it calls Mininet initialization functions.
+
+    # Starting topology, create tmp files which are loaded to routers
+    #  to start deamons and then start routers
+    start_topology(tgen)
+
+    # Creating configuration from JSON
+    build_config_from_json(tgen, topo)
+
+    input_dict = {
+        "r1": {"bgp": {"local_as": "100", "vrf": "GREEN", "router_id": "1.1.1.1"}},
+        "r2": {"bgp": {"local_as": "200", "vrf": "GREEN", "router_id": "2.2.2.2"}}
+    }
+    result = create_router_bgp(tgen, topo, input_dict)
+    assert result is True, "Testcase {} :Failed \n Error: {}".format(tc_name, result)
+
+    step("configure update-source on the interface")
+    raw_config = {
+        "r1": {
+            "raw_config": [
+                "ipv6 forwarding",
+                "router bgp {} vrf {}".format(topo["routers"]["r1"]["bgp"][0]["local_as"], topo["routers"]["r1"]["vrfs"][0]["name"]),
+                "no neighbor {}".format(topo["routers"]["r2"]["links"]["r1"]["ipv4"].split("/")[0]),
+                "no neighbor {}".format(topo["routers"]["r2"]["links"]["r1"]["ipv6"].split("/")[0]),
+                "neighbor {} interface remote-as {}".format("r1-r2-eth0", topo["routers"]["r2"]["bgp"][0]["local_as"]),
+                "address-family ipv6 unicast",
+                "neighbor {} activate".format("r1-r2-eth0"),
```

Comment:
Shouldn't the peering be numbered? fe80::1 with fe80::2? As I understand this is the case you want to cover and not the unnumbered stuff.

Reply:
I had raised this issue in the BGP group in slack and @donaldsharp and others had suggested this config for testing link-local.

Reply:
I believe this test does not test IPv6 link-local under VRF peering as defined in the commit. Or without this change unnumbered peering does not work as well with IPv6?

Reply:
Hi @ton31337 , I checked that without this change also, ipv6 link-local peering does not work. I tried using the existing testcase bgp_vrf_lite_ipv6_rtadv. If we use normal ipv6 address, it passes.  I believe there may be an issue with mininet infrastructure for this. The testcase passes in the actual box, but not in the virtual environment.

Reply:
>The testcase passes in the actual box, but not in the virtual environment.

Then what is the real point of testing it here if it's not working?

Reply:
I suspect that I might be missing some linux parameters which are needed for binding ipv6 link-local address to the BGP socket. 
Also, I see that there are no ipv6 link-local testcases in BGP topotest. Can you refer me to some of the already existing link-local testcases in BGP topotest.

Reply:
Yes, I don't see arbitrary IPv6 link-local addresses in topotests. Maybe @davischw can help us here? @davischw is it possible to use custom IPv6 link-local addresses between neighbors in topotests?

Reply:
@davischw : I am kind of stuck here.  can you tell me why my testcase in this pull request is not passing. Or can you point me to any BGP testcase which tests link-local address as a statically configured neighbor.

Reply:
@sudhanshukumar22 & @ton31337

Reply:
@davischw overall, is it possible to use custom LL addresses for peerings? Not unnumbered stuff, but the peering using just a LL addresses.